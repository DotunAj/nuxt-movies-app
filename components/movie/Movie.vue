<template>
  <section class="right-section">
    <div class="mv-description">
      <h3>{{ movie.original_title }}</h3>
      <p class="mv-genre">{{ movie.genre }}</p>
      <div class="foo">
        <p class="mv-rating">
          ðŸ”¥
          {{ movie.vote_average }}<span>{{ movie.vote_count }}</span>
        </p>
        <p class="mv-year">
          <span>{{ movie.releaseYear }}</span
          >{{ movie.runtime }}min
        </p>
      </div>
    </div>
    <div class="mv-synopsis">
      <p>
        {{ movie.overview }}
      </p>
    </div>
    <div class="mv-images">
      <div class="mv-image image-1">
        <img :src="movie.posterUrl" alt="ps-1" />
      </div>
      <div class="mv-image image-2">
        <img :src="backdrops[1].backdropUrl" alt="ps-1" />
      </div>
      <div class="mv-image image-3">
        <img :src="backdrops[2].backdropUrl" alt="ps-1" />
      </div>
      <div class="mv-image image-4">
        <img :src="backdrops[3].backdropUrl" alt="ps-1" />
      </div>
    </div>
  </section>
</template>

<script>
export default {
  props: {
    movie: {
      type: Object,
      default: () => {}
    },
    backdrops: {
      type: Array,
      default: () => {}
    }
  }
}
</script>
