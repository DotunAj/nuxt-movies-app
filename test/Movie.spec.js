import { shallowMount } from '@vue/test-utils'
import Movie from '~/components/movie/Movie'

const movie = {
  popularity: 1,
  posterUrl: 'mybackdrop.com/image1',
  original_title: 'A new Hope',
  releaseYear: 2019,
  runtime: 170,
  genre: 'Action',
  vote_average: 7.0,
  vote_count: 500,
  overview: 'lorem ipsum yada yada yada'
}

const backdrops = [
  {
    backdropUrl: 'img.jpg'
  },
  {
    backdropUrl: 'img.jpg'
  },
  {
    backdropUrl: 'img.jpg'
  },
  {
    backdropUrl: 'img.jpg'
  }
]

describe('Movie', () => {
  test('it should render the correct content', () => {
    const wrapper = shallowMount(Movie)
    wrapper.setProps({ movie, backdrops })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
