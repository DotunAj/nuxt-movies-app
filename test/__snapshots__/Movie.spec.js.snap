// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie it should render the correct content 1`] = `
"<section class=\\"right-section\\"><div class=\\"mv-description\\"><h3>A new Hope</h3> <p class=\\"mv-genre\\">Action</p> <div class=\\"foo\\"><p class=\\"mv-rating\\">
        ðŸ”¥
        7<span>500</span></p> <p class=\\"mv-year\\"><span>2019</span>170min
      </p></div></div> <div class=\\"mv-synopsis\\"><p>
      lorem ipsum yada yada yada
    </p></div> <div class=\\"mv-images\\"><div class=\\"mv-image image-1\\"><img alt=\\"ps-1\\" src=\\"mybackdrop.com/image1\\"></div> <div class=\\"mv-image image-2\\"><img src=\\"img.jpg\\" alt=\\"ps-1\\"></div> <div class=\\"mv-image image-3\\"><img src=\\"img.jpg\\" alt=\\"ps-1\\"></div> <div class=\\"mv-image image-4\\"><img src=\\"img.jpg\\" alt=\\"ps-1\\"></div></div></section>"
`;
